modules/network-observability-flowcollector-api-specifications.adoc

```diff
@@ -2813,6 +2949,131 @@ Fields absent from the 'k8s.v1.cni.cncf.io/network-status' annotation must not b
 | `string`
 | `name` should match the network name as visible in the pods annotation 'k8s.v1.cni.cncf.io/network-status'.
 
+|===
+== .spec.processor.deduper
+Description::
++
+--
+`deduper` allows to sample or drop flows identified as duplicates, in order to save on resource usage.
+IMPORTANT: This feature is available as a Developer Preview.
+--
+
+Type::
+  `object`
+
+
+
+
+[cols="1,1,1",options="header"]
+|===
+| Property | Type | Description
+
+| `mode`
+| `string`
+| Set the Processor de-duplication mode. It comes in addition to the Agent-based deduplication because the Agent cannot de-duplicate same flows reported from different nodes. +
+
+- Use `Drop` to drop every flow considered as duplicates, allowing saving more on resource usage but potentially loosing some information such as the network interfaces used from peer, or network events. +
+
+- Use `Sample` to randomly keep only 1 flow on 50 (by default) among the ones considered as duplicates. This is a compromise between dropping every duplicates or keeping every duplicates. This sampling action comes in addition to the Agent-based sampling. If both Agent and Processor sampling are 50, the combined sampling is 1:2500. +
+
+- Use `Disabled` to turn off Processor-based de-duplication. +
+
+
+| `sampling`
+| `integer`
+| `sampling` is the sampling rate when deduper `mode` is `Sample`.
+
+|===
+== .spec.processor.filters
+Description::
++
+--
+`filters` let you define custom filters to limit the amount of generated flows.
+IMPORTANT: This feature is available as a Developer Preview.
+--
+
+Type::
+  `array`
+
+
+
+
+== .spec.processor.filters[]
+Description::
++
+--
+`FLPFilterSet` defines the desired configuration for FLP-based filtering satisfying all conditions
+--
+
+Type::
+  `object`
+
+
+
+
+[cols="1,1,1",options="header"]
+|===
+| Property | Type | Description
+
+| `allOf`
+| `array`
+| `filters` is a list of matches that must be all satisfied in order to remove a flow.
+
+| `outputTarget`
+| `string`
+| If specified, this filters only target a single output: `Loki`, `Metrics` or `Exporters`. By default, all outputs are targeted.
+
+| `sampling`
+| `integer`
+| `sampling` is an optional sampling rate to apply to this filter.
+
+|===
+== .spec.processor.filters[].allOf
+Description::
++
+--
+`filters` is a list of matches that must be all satisfied in order to remove a flow.
+--
+
+Type::
+  `array`
+
+
+
+
+== .spec.processor.filters[].allOf[]
+Description::
++
+--
+`FLPSingleFilter` defines the desired configuration for a single FLP-based filter
+--
+
+Type::
+  `object`
+
+Required::
+  - `field`
+  - `matchType`
+
+
+
+[cols="1,1,1",options="header"]
+|===
+| Property | Type | Description
+
+| `field`
+| `string`
+| Name of the field to filter on
+Refer to the documentation for the list of available fields: https://docs.openshift.com/container-platform/latest/observability/network_observability/json-flows-format-reference.html.
```

Comment:
No links to OCP docs in a module.

Reply:
I think Joel changed this to github link since that's allowed in the exception for autogenerated API docs. I can add the xref to the downstream here, but since this is autogenerated from the upstream, I think that's why there was the link to the downstream docs.

Reply:
NVM the module can't accept the xref 🤦‍♀️ just reverted to the GitHub link since at least we have an exception for that in autogenerated docs.